proposals/XXXX-swift-archival-serialization.md

```diff
@@ -0,0 +1,2226 @@
+    }
+}
+```
+
+Unkeyed containers allow for the same types of nesting:
+
+```swift
+// Continuing from before
+public protocol UnkeyedEncodingContainer {
+    /// Encodes a nested container keyed by the given type and returns it.
+    ///
+    /// - parameter keyType: The key type to use for the container.
+    /// - returns: A new keyed encoding container.
+    mutating func nestedContainer<NestedKey : CodingKey>(keyedBy keyType: NestedKey.Type) -> KeyedEncodingContainer<NestedKey>
+
+    /// Encodes an unkeyed encoding container and returns it.
+    ///
+    /// - returns: A new unkeyed encoding container.
+    mutating func nestedUnkeyedContainer() -> UnkeyedEncodingContainer
+}
+
+public protocol UnkeyedDecodingContainer {
+    /// Decodes a nested container keyed by the given type.
+    ///
+    /// - parameter type: The key type to use for the container.
+    /// - returns: A keyed decoding container view into `self`.
+    /// - throws: `CocoaError.coderTypeMismatch` if the encountered stored value is not a keyed container.
+    mutating func nestedContainer<NestedKey : CodingKey>(keyedBy type: NestedKey.Type) throws -> KeyedDecodingContainer<NestedKey>
+
+    /// Decodes an unkeyed nested container.
+    ///
+    /// - returns: An unkeyed decoding container view into `self`.
+    /// - throws: `CocoaError.coderTypeMismatch` if the encountered stored value is not an unkeyed container.
+    mutating func nestedUnkeyedContainer() throws -> UnkeyedDecodingContainer
+}
+```
+
+### Dynamic Context-Based Behavior
+
+In some cases, types may need context in order to decide on their external representation. Some types may choose a different representation based on the encoding format that they are being read from or written to, and others based on other runtime contextual information. To facilitate this, `Encoders` and `Decoders` expose user-supplied context for consumption:
+
+```swift
+/// Represents a user-defined key for providing context for encoding and decoding.
+public struct CodingUserInfoKey : RawRepresentable, Hashable {
+    typealias RawValue = String
+    let rawValue: String
+    init?(rawValue: String)
+}
+
+// Continuing from before:
+public protocol Encoder {
+    /// Any contextual information set by the user for encoding.
+    var userInfo: [CodingUserInfoKey : Any] { get }
+}
+
+public protocol Decoder {
+    /// Any contextual information set by the user for decoding.
+    var userInfo: [CodingUserInfoKey : Any] { get }
+}
+```
+
+Consuming types may then support setting contextual information to inform their encoding and decoding:
+
+```swift
+public struct Person : Encodable {
+    public static let codingUserInfoKey = CodingUserInfoKey(rawValue: "com.foocorp.person.codingUserInfoKey")
+
+    public struct UserInfo {
+        let shouldEncodePrivateFields: Bool
+        // ...
+    }
+
+    func encode(to encoder: Encoder) throws {
+        if let context = encoder.userInfo[.codingUserInfoKey] as? Person.UserInfo {
```

Comment:
Good catches, thanks! This was a holdover from a previous iteration that didn't get updated properly. Will fix.